Include/unicodeobject.h

```diff
@@ -465,6 +465,24 @@ PyAPI_FUNC(PyObject*) PyUnicode_AsUTF8String(
     PyObject *unicode           /* Unicode object */
     );
 
+/* Returns a pointer to the default encoding (UTF-8) of the
+   Unicode object unicode and the size of the encoded representation
+   in bytes stored in *size.
+
+   In case of an error, no *size is set.
+
+   This function caches the UTF-8 encoded string in the unicodeobject
+   and subsequent calls will return the same string.  The memory is released
+   when the unicodeobject is deallocated.
+
+   _PyUnicode_AsStringAndSize is a #define for PyUnicode_AsUTF8AndSize to
+   support the previous internal function with the same behaviour.
+*/
+
+PyAPI_FUNC(const char *) PyUnicode_AsUTF8AndSize(
+    PyObject *unicode,
+    Py_ssize_t *size);
+
```

Comment:
```c
#if !defined(Py_LIMITED_API) || Py_LIMITED_API+0 >= 0x030A0000
```

@vstinner Can we add this with `Py_LIMITED_API+0 >= 0x03030000` because we had this API from before Python 3.3?

Reply:
I wanted to, because it would make my life easier, but I don't think it'll work on Windows (since the symbols go in a different DLL), so I think it needs to require a higher `Py_LIMITED_API`

Reply:
I'll wait for @vstinner's response before I add it.

Reply:
@serhiy-storchaka knows the details about the stable ABI. I don't know.

Reply:
First, I think it would not work on Windows, because the name is not exported in older Python releases. Second, it may not work on third-party Python implementations which did not include it in the C API set.